app/helpers/repository_helper.rb

```diff
@@ -136,6 +136,13 @@ def add_folder(folder_path, user, repo, path: '/', txn: nil, required_files: nil
     # check if only required files are allowed for a submission
     # allowed folders = paths in required files
     if required_files.present? && required_files.none? { |file| file.starts_with?(folder_path) }
+      folder_path = format_folder_path folder_path
+      messages << [:invalid_folder_name, folder_path]
+      return false, messages
+    end
+
+    if folder_path.include?('/.git/')
```

Comment:
Use [`each_filename`](https://ruby-doc.org/3.3.0/exts/pathname/Pathname.html#method-i-each_filename) for more precise checking for a `.git` segment